### ISSUE TITLE
Cannot generate a 5xx exception when response body is null

### ISSUE THREAD

**Issue by mlambley** [elastic/elasticsearch-php#1054]
### Summary of problem or feature request
Cannot generate a 5xx exception when response body is null. Error shown:
```
Wrong parameters for Elasticsearch\Common\Exceptions\ServerErrorResponseException([string $message [, long $code [, Throwable $previous = NULL]]]) in elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php:667
```

### Code snippet of problem
Happens on my Magento 2.4 installation when `bin/magento indexer:reindex`
https://github.com/elastic/elasticsearch-php/blob/da40f294cbffaea61729456746ecba14310eafdc/src/Elasticsearch/Connections/Connection.php#L665

### System details
- Operating System Ubuntu 20
- PHP Version 7.3
- ES-PHP client version dev-master da40f29
- Elasticsearch version 7.8.1

You've fixed the error here in `process4xxError`
https://github.com/elastic/elasticsearch-php/blob/da40f294cbffaea61729456746ecba14310eafdc/src/Elasticsearch/Connections/Connection.php#L616-L619

Need to apply the same fix to `process5xxError`. But may I suggest something else. My attached PR will make both 5xx and 4xx show a generic error message rather than just "null". That way, when Magento reports "Unknown error" it will give details about the fact that it was my elasticsearch server configuration that's the problem.

Thank you.

**Comment by mlambley** [comment]
PR submitted https://github.com/elastic/elasticsearch-php/pull/1055

**Comment by SeBsZ** [comment]
@mlambley thanks for this. I just ran into this same issue when 3 out of 4 ES servers were down. I got the `Wrong parameters for Elasticsearch\Common\Exceptions\ServerErrorResponseException` in `process5xxError()`. 

In my case, just like for @mlambley, `$responseBody` is `null`, and that is not allowed by 
`$exception = new ServerErrorResponseException($responseBody, $statusCode);`

I quickly glanced your PR and it looks fine to me, though I did not test it yet. Hopefully @ezimuel will have time to look into this soon so it can make it into a release.

**Comment by ezimuel** [comment]
@mlambley thanks for PR #1055. I'll look into it asap!

**Comment by archon810** [comment]
👀 Stuff seems to be crashing because of this. Thanks for looking into it.

**Comment by mlambley** [comment]
Thanks. My only concern is that I won't actually get the changes if the PR is accepted as is, because I'm using Magento which hardcodes an earlier version of elastic. I'm looking for guidance regarding which branch to target. I'm thinking `6.x`

**Comment by archon810** [comment]
@ezimuel Any idea if the fix will make it into the next release yet? Thanks.

**Comment by ezimuel** [comment]
@mlambley I just sent a review about PR #1055. 
@archon810 We need to finalize the PR, this will require some time since I asked for a change. I think we can release in a patch release after the next minor `7.9` that is scheduled this week.

**Comment by mlambley** [comment]
New PR https://github.com/elastic/elasticsearch-php/pull/1056

**Comment by archon810** [comment]
Any updates here?

**Comment by mlambley** [comment]
@archon810 @ezimuel https://github.com/elastic/elasticsearch-php/pull/1056 is ready for review. Thank you.

**Comment by ezimuel** [comment]
Merged in #1056.
